### PR TITLE
Change default Grafana to v10

### PR DESCRIPTION
**Pull request by Regrau**
## Description

This PR updates our Grafana deployment by making Grafana version 10 the default version accessible on the main domain. As part of this update:
- Grafana version 10 will now be available on the primary domain.
- The previous Grafana version 8 instance will be moved to a secondary domain for users who need more time to migrate.

### Why do we need it, and what problem does it solve?

This is the second phase of our three-step migration process to Grafana 10:
- **Default Version**: Establishes Grafana 10 as the default version, making it accessible on the main domain.
- **Encourages Adoption**: Promotes the use of Grafana 10, which offers enhanced performance, new features, and improved security.
- **Supports Transition**: Allows users who need more time to migrate to continue using Grafana 8 on a secondary domain, ensuring operational continuity.
- **Minimizes Disruption**: This intermediary step ensures users transition smoothly by maintaining access to both versions during the migration period.

The final phase will involve the complete deprecation of Grafana 8, so this step is crucial for ensuring that all users are aware of the transition and have adequate time to fully migrate their workflows.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: chore
summary: Made Grafana version 10 the default on primary domain, moved Grafana 8 to secondary domain.
impact_level: default 
```

### CODE REVIEW

**Comment by vladimirGuryanov** [comment]
You forgot to change the Grafana domain here: https://github.com/deckhouse/deckhouse/pull/9076/files#diff-3e2e027c44bdea446fe55ebae34e2f53060863a63fa41b796db547322eec8dcaR35

**Review by vladimirGuryanov — CHANGES_REQUESTED** [review]

**Review by vladimirGuryanov — APPROVED** [review]

File: modules/300-prometheus/templates/grafana/ingress-v10.yaml [hunk 1]
```diff
@@ -32,19 +35,19 @@ metadata:
...
+        rewrite ^ {{ include "helm_lib_module_uri_scheme" . }}://{{ include "helm_lib_module_public_domain" (list . "grafana") }} permanent;
       }
     nginx.ingress.kubernetes.io/affinity: cookie
 spec:
   ingressClassName: {{ include "helm_lib_module_ingress_class" . | quote }}
   {{- if (include "helm_lib_module_https_ingress_tls_enabled" .) }}
   tls:
   - hosts:
-    - {{ include "helm_lib_module_public_domain" (list . "grafana-v10") }}
+    - {{ include "helm_lib_module_public_domain" (list . "grafana") }}
```
Comment (vladimirGuryanov) [inline modules/300-prometheus/templates/grafana/ingress-v10.yaml:46]: Need to add `- {{ include "helm_lib_module_public_domain" (list . "prometheus") }}` and remove it from ingress.yaml.

**Review by vladimirGuryanov — CHANGES_REQUESTED** [review]

**Review by vladimirGuryanov — APPROVED** [review]

**Review by lazovskiy — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10141569559 10141569559-1-con-1-27 /sys/deckhouse-oss/install:pr9076 redos@89.169.137.246

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'e2e/run/gcp'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10142066574 10142066574-1-con-1-27 /sys/deckhouse-oss/install:pr9076 redos@89.169.143.95

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**